packages/cli-kit/src/public/node/fs.ts

```diff
@@ -440,6 +442,32 @@ export async function fileHasExecutablePermissions(path: string): Promise<boolea
   }
 }
 
+export function fileHasWritePermissions(path: string): boolean {
+  try {
+    accessSync(path, fsConstants.W_OK)
+    return true
+    // eslint-disable-next-line no-catch-all/no-catch-all
+  } catch {
+    return false
+  }
+}
+
+export function fileIsOwnedByCurrentUser(path: string): boolean {
+  // process.getuid() is only available on Unix-like systems
+  if (process.platform === 'win32' || typeof process.getuid !== 'function') return true
```

Comment:
We shouldn't return true in this case. What if we return undefined instead?

I've checked that there's a way to find it out for Windows, but:
- All the Observe errors happen in Mac
- In Windows is not very common to have this kind of permission issues with files

So I agree we can skip it.